### PR TITLE
Seb adjustable accounts for stats

### PR DESCRIPTION
**Pull request by sdl433**
This PR gives the user the functionality to choose the accounts for which they can view stats. I also implemented much needed design changes, as well as refactored some code to make things more organized. Please review, let me know your thoughts, as well as if anything is needed
Screenshot 2021-12-05 203930
Screenshot 2021-12-05 204641

Screenshot 2021-12-05 204021
Screenshot 2021-12-05 204108

.

### CODE REVIEW

File: front-end/src/components/Statistics/utils/prepare_stats.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+ * Gathers the spending and balance trends for the statistics
+ * @param {*} days  The number of days the stats elapse
+ * @param {*} accountToBalance  A mapping from the account's name to the account's balance
+ * @param {*} dateToMoneySpent  A mapping from the date to the amount of money spent for the accounts selected
+ * @param {*} dateToNet  A mapping from the date to the net amount of money gained in the selected accounts' total balance
+ * @param {*} selectedAccounts  An object indicating which accounts were selected
+ */
+const getTrends = (days, accountToBalance, dateToMoneySpent, dateToNet, selectedAccounts=null) => {
+  let balanceTrend = [];
+  let spendingTrend = [];
```
Comment (azc242) [inline front-end/src/components/Statistics/utils/prepare_stats.js:94]: Can we turn all the `let` into something more robust? I don't think we're supposed to use `let`.

File: front-end/src/components/css/StatsNavbar.css [hunk 2]
```diff
@@ -19,6 +19,16 @@
   background-color: #1a83ff;
 }
 
+.statsNavLinkActive {
+  display: flex;
+  float: right;
+  color: white;
+  text-align: center;
+  padding: 14px 16px;
```
Comment (azc242) [inline front-end/src/components/css/StatsNavbar.css:27]: nit: can you try to use dynamic units? I'm worried this might break on mobile or different sized screens

File: front-end/src/components/Statistics/Charts/PieChart.js [hunk 3]
```diff
@@ -28,6 +28,7 @@ const PieChart = ({ name, data }) => {
           width: "100%",
           height: "100%",
         },
+        backgroundColor: { fill: 'transparent' }
```
Comment (azc242) [inline front-end/src/components/Statistics/Charts/PieChart.js:31]: I really like this, it looks super slick!

File: front-end/src/components/css/Timebar.css [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  background-color: #060b26;
+}
+
+.timeOption {
+  display: flex;
+  float: left;
+  color: white;
+  text-align: center;
+  width: 243px;
+  padding: 14px 16px;
```
Comment (azc242) [inline front-end/src/components/css/Timebar.css:15]: same as above, try not to use fixed sizes for responsiveness

**Review by azc242 — CHANGES_REQUESTED** [review]
The changes look great (love the transparent background on the charts)! 
Just a few comments/changes for best practices (not using `let` and not using magic numbers for `px`).

**Comment by sdl433** [comment]
Hey, I took a look at the suggested changes and implemented them. Please take a look and review again, and please suggest any other modifications that come to mind.

File: front-end/src/components/Statistics/utils/prepare_stats.js (hunk 1 above)
Comment (sdl433) [reply front-end/src/components/Statistics/utils/prepare_stats.js]: I changed all the `let`s in `prepare_stats.js` into `consts`, except for `balance`

File: front-end/src/components/css/StatsNavbar.css (hunk 2 above)
Comment (sdl433) [reply front-end/src/components/css/StatsNavbar.css]: I just changed those constants into percentages.

File: front-end/src/components/Statistics/Charts/PieChart.js (hunk 3 above)
Comment (sdl433) [reply front-end/src/components/Statistics/Charts/PieChart.js:31]: Thanks, I think so too!

File: front-end/src/components/css/Timebar.css (hunk 4 above)
Comment (sdl433) [reply front-end/src/components/css/Timebar.css]: Already fixed this issue using percents as well.

**Review by tangym27 — COMMENTED** [review]
this looks really beautiful! nice job woo 

three minor things:
1 - if i dont spend anything (#broke life) this is what the chart looks like:
Screen Shot 2021-12-08 at 5 25 37 PM
is this intended?

2 - (style)
Screen Shot 2021-12-08 at 5 35 52 PM
i think it might look better if you switched the positions of the navbars. balance and spending seems 'higher' than the date features. as the balance and spending are the tabs themselves and that bar covers the whole screen so it makes sense for it to be close to the top. then the navbar that sorts by date should be a subsection of that (it has a caption of `The following charts` so it feels weird that immediately following is a navbar). Also looking at the date thing makes me notice all the negative space of the original navbar so it might be nice to shorten it.

tldr: i think this looks a bit nicer but wanna hear your thoughts
image

3- stats kinda take a long time to load for me. should i be doing something to make it faster?

**Comment by sdl433** [comment]
I made a commit just now.

Addressing 1: The Spending page should now show that there is "No Spending Data Available", please confirm that that is indeed the case.

Addressing 2: I implemented your suggestion

Addressing 3: Ah yes, stats currently takes a bit of a long time, especially if we're loading data from the last year. There's nothing the user could do to make it faster.

File: front-end/src/components/css/StatsNavbar.css [hunk 5]
```diff
@@ -1,24 +1,41 @@
...
+.statsNavLinkActive {
+  display: flex;
+  float: left;
+  color: white;
+  text-align: center;
+  width: 50%;
+  padding: 2% 2%;
+  text-decoration: none;
+  background-color: #1a83ff;
+}
```
Comment (tangym27) [inline front-end/src/components/css/StatsNavbar.css:37]:
```suggestion
justify-content: center;
}
```

File: front-end/src/components/css/StatsNavbar.css [hunk 6]
```diff
@@ -1,24 +1,41 @@
...
   display: flex;
-  float: right;
+  float: left;
   color: white;
   text-align: center;
-  padding: 14px 16px;
+  width: 50%;
+  padding: 2% 2%;
   text-decoration: none;
 }
```
Comment (tangym27) [inline front-end/src/components/css/StatsNavbar.css:22]:
```suggestion
justify-content: center;
}
```

File: front-end/src/components/css/Timebar.css [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+.timeOption {
+  display: flex;
+  float: left;
+  color: white;
+  text-align: center;
+  width: 20%;
+  padding: 2% 2%;
+  text-decoration: none;
+}
```
Comment (tangym27) [inline front-end/src/components/css/Timebar.css:17]:
```suggestion
justify-content: center;
}
```

File: front-end/src/components/css/Timebar.css [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+.timeOptionActive {
+  display: flex;
+  float: left;
+  color: white;
+  text-align: center;
+  width: 20%;
+  padding: 2% 2%;
+  text-decoration: none;
+  background-color: #1a83ff;
+}
```
Comment (tangym27) [inline front-end/src/components/css/Timebar.css:28]:
```suggestion
justify-content: center;
}
```

File: back-end/app.js [hunk 9]
```diff
@@ -65,6 +65,9 @@ let ITEM_ID = null;
 // persistent data store
 let PAYMENT_ID = null;
 
+// REMOVE
+process.env.DB_URL = "mongodb+srv://sneaky-seven:sneaky-seven@cluster0.6ophh.mongodb.net/database?retryWrites=true&w=majority"
+
```
Comment (tangym27) [inline back-end/app.js:70]:
```suggestion
```

I assume you want to remove it?

**Review by tangym27 — COMMENTED** [review]
Left five commit suggestions.

First one was towards the removal of db secrets
Other four was so that the stat navbar was centered to look like this:
image

If you prefer the other way and want them aligned to the left, feel free to reject the suggestions (centering looks nice to me but I will trust your judgement).

Either way, once you resolve these commits, lmk and I'll approve.

**Comment by sdl433** [comment]
Haha sorry about revealing the db secret, I keep on forgetting to remove it. As for the other suggestions, I just added them. Please let me know if anything else is needed.

**Review by tangym27 — APPROVED** [review]
Thanks for changing! I don't have multiple accounts to test it on but I assume you tested it too.

Stats look great :D!
